### ISSUE TITLE
Manually set mass

### ISSUE THREAD

**Issue by stargazing-dino** [Jondolf/bevy_xpbd#50]
I'm messing around with some simple planets and I'm trying to manually set the mass in order to calculate gravitational pull or what not. Right now, I can do something like that:

```rs
#[derive(Bundle)]
pub struct PlanetBundle {
    pub planet: Planet,

    pub position: Position,

    pub rigid_body: RigidBody,

    pub mass: Mass,
}
```

But it's never exactly what I set it to after the first frame or so. For example, when I set a mass of `200.0` it eventually gets changed to `219.31` or something. I remember seeing some code in the lib that added to the mass based off the mass properties so that's probably why.

Is there any way to override the default behavior of the mass changing?

**Comment by Jondolf** [comment]
Do the planets have colliders? If they do, then the mass properties of the colliders are computed and added on top of the body's existing mass properties, like in rapier. This is documented [here](https://docs.rs/bevy_xpbd_3d/latest/bevy_xpbd_3d/components/enum.RigidBody.html#adding-mass-properties).

You can set the mass properties of the colliders to zero by adding `ColliderMassProperties::ZERO`. With that I am not seeing any mass changing in my examples.

**Comment by stargazing-dino** [comment]
That worked great for me ! Thank you. I'm enjoying the library :)

**Comment by stargazing-dino** [comment]
tldr: Nvm, I'm got it :P

I feel like I'm silly for doing this but if I try to do the same thing of setting `ColliderMassProperties::ZERO` with a `RigidBody::Dynamic` it doesn't work and my little ball disappears of the screen before I've even added any forces to the ball.

```rs
let shape = shape::UVSphere {
    radius: 1.0,
    ..default()
};

commands.spawn((
    PbrBundle {
        mesh: meshes.add(shape.into()),
        material: materials.add(Color::rgb(0.8, 0.7, 0.6).into()),
        transform: Transform::from_xyz(0.0, 0.0, 0.0),
        ..default()
    },
    Collider::ball(1.0),
    ColliderMassProperties::ZERO,
    // RigidBody::Kinematic, // <- Works
    // RigidBody::Dynamic, // <- hops off the screen before I've started the physics 
    Position(Vector::new(14.0, 0.0, 0.0)),
    ExternalForce::default(),
    Mass(1.0),
    Friction::new(6.0),
    Junk {},
));
```

Again, I'm crazy dumb when it comes to physics anything so feel free to let me know if I'm just doing stuff wrong.

# Edit

So I guess it just needs a non-zero density in order to not freak out:

```rs
ColliderMassProperties {
    density: 1.0,
    ..default()
},
Mass(1.0),
RigidBody::Dynamic,
```

Which makes some sense. Anyways, I guess the problem now is that the mass is changing from the `1.0` I set it to to something like `6.4` 😭

# Edit 2

I guess setting the density to something super low actually fixes it and I get my mass of ~1.0 lol.


```rs
ColliderMassProperties {
    density: 0.00001, // honestly, just non-zero prob
    ..default()
},
```

**Comment by Jondolf** [comment]
So the problem isn't actually the non-zero density collider, it's that you only add `Mass` and not `Inertia` as well.

Angular inertia controls how much effort it requires to rotate a body, and when there's zero inertia, there is probably some kind of division by zero happening, which breaks the simulation and causes the bodies to accelerate into infinity (making them disappear).

So I would add something like this:

```rust
(
    ColliderMassProperties::ZERO,
    Mass(1.0),
    Inertia(1.0), // In 2D
    Inertia(Mat3::IDENTITY) // In 3D
    // ...other components
)
```

You can of course set the inertia to whatever you want, as long as it's greater than 0.

Thanks for the issue though, I definitely need to make the docs more explicit about this. I might also just add a warning when the engine detects a dynamic body with zero mass/inertia.

**Comment by Jondolf** [comment]
Also FYI, it looks like you might be using the crate's math types like `Vector`. This is fine, but you can also just use the normal types like `Vec2` or `Vec3`. The special math types are there because the crate needs to use different types depending on feature flags like `2d`/`3d` and `f32`/`f64`, but in actual usage you already know the types because you have specified the features.
